libs/ui/src/hooks/useHardware.test.ts

```diff
@@ -0,0 +1,310 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+import { Logger, LogSource, LogEventId } from '@votingworks/logging';
+import {
+  AccessibleControllerProductId,
+  AccessibleControllerVendorId,
+  FujitsuFi7160ScannerProductId,
+  FujitsuScannerVendorId,
+  MemoryHardware,
+  OmniKeyCardReaderProductId,
+  OmniKeyCardReaderVendorId,
+  PlustekScannerVendorId,
+  PlustekVTM300ScannerProductId,
+} from '@votingworks/utils';
+import { useHardware } from './useHardware';
+
+test('can connect printer as expected', async () => {
+  const hardware = new MemoryHardware();
+  const fakeLogger = new Logger(LogSource.VxBatchScanApp);
+  const logSpy = jest.spyOn(fakeLogger, 'log').mockResolvedValue();
+  const { result, rerender } = renderHook(() =>
+    useHardware({ hardware, logger: fakeLogger })
+  );
+  expect(result.current.hasPrinterAttached).toBe(false);
+  expect(logSpy).toHaveBeenCalledTimes(0);
+
+  await act(async () => await hardware.setPrinterConnected(true));
+  rerender();
+  expect(result.current.hasPrinterAttached).toBe(true);
+  expect(logSpy).toHaveBeenCalledTimes(2);
+  expect(logSpy).toHaveBeenNthCalledWith(
+    1,
+    LogEventId.PrinterAttached,
+    'system',
+    expect.objectContaining({
+      message: expect.stringContaining('New printer attached'),
+      connected: 'true',
```

Comment:
Why a `string` value here (and for `vendorId` etc elsewhere)?

Reply:
I had typed the logging function to only take strings, but that seems silly now so I'll update to also accept boolean/numbers.